pkg/cloud/services/eks/nodegroup.go

```diff
@@ -588,6 +591,29 @@ func (s *NodegroupService) setStatus(ng *eks.Nodegroup) error {
 		managedPool.Status.Ready = false
 	case eks.NodegroupStatusUpdating:
 		managedPool.Status.Ready = true
+	case eks.NodegroupStatusDegraded:
+		issueErrMsgSet := make([]string, 0)
+		var errMsgStr string
+
+		for _, iss := range ng.Health.Issues {
+			errMsg := iss.GoString()
+			if slices.Contains(issueErrMsgSet, errMsg) {
+				continue
+			}
+			issueErrMsgSet = append(issueErrMsgSet, errMsg)
+			errMsgStr = fmt.Sprintf("%s %s", errMsgStr, errMsg)
+		}
+		reason := capierrors.InvalidConfigurationMachineError
+		// TODO: implement checks for other MachineStatusErrors and set reason accordingly
```

Comment:
Are you planning to add this TODO as part of this PR?

Reply:
Hi @Ankitasw I was going to leave this TODO as this PR is implementing returning of the error set.